controllers/triggerservice/triggerservice_controller.go

```diff
@@ -91,25 +92,36 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, err
 	}
 
-	deploymentList := &appsv1.DeploymentList{}
-	if err := r.List(ctx, deploymentList, client.MatchingLabels(map[string]string{
-		triggerNameLabel: ts.Name,
-	})); err != nil {
+	if err := r.handleWorker(ctx, ts); err != nil {
+		logger.Error(err, "failed to handle trigger worker")
 		return ctrl.Result{}, err
 	}
-	// if no worker deployment exists, create a new one to run trigger
-	if len(deploymentList.Items) == 0 {
-		if err := r.createWorker(ctx, ts); err != nil {
-			logger.Error(err, "failed to create worker deployment for trigger")
-			return ctrl.Result{}, err
-		}
-		logger.Info("successfully create worker deployment for trigger")
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) handleWorker(ctx context.Context, ts *standardv1alpha1.TriggerService) error {
+
+	v, err := r.loadTemplateCueValue(ctx, ts)
+	if err != nil {
+		return err
 	}
 
-	return ctrl.Result{}, nil
+	if err := r.createRoles(ctx, ts, v); err != nil {
+		return err
+	}
+
+	if err := r.createDeployment(ctx, ts, v); err != nil {
+		return err
+	}
+
+	if err := r.createService(ctx, ts, v); err != nil {
```

Comment:
Please check `createService` first, if the value is true, then create the service.

Reply:
you mean check before invoking the function? or we can check the value in the function ?